File: twoSum/twoSum.js
var twoSum = function(nums, target) {
    let map = new Map();
    
    for(let i = 0; i < nums.length; i ++) {
        let base = nums[i];
        if(map.has(base)) {
            return [map.get(base), i];
        } else {
            map.set(target - base, i);
        }
    }
	return [];
};

File: twoSum/twoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        compliments = {}
        for i,n in enumerate(nums):
            base = nums[i]
            if(base in compliments):
                return [compliments[base], i]
            compliments[target - base] = i
        return